test/routes/user_spec.rb

```diff
@@ -1,8 +1,26 @@
+require 'pry'
 require_relative '../spec_helper.rb'
 require_relative '../../app.rb'
+require_relative '../../models/user.rb'
+require_relative '../../models/user_category.rb'
 
 RSpec.describe App  do
 
+  before(:context) do
+    User.delete_all
+    UserCategory.delete_all
+  end
+
+  it 'should add user user to the database' do
+    post '/user/signup', params={ user: { username: "naruto", email: "naruto@email.com", password: "tester123" } }
+
+    expect(last_response).to be_redirect
+    follow_redirect!
+
+    expect(last_response).to be_ok
+    expect(last_response.body).to include("Customize your learning")
+  end
+
```

Comment:
Add sad tests for the happy test above. Eg, `it should not add user if params are invalid`